sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -236,6 +236,8 @@ case class HashAggregateExec(
          | }
        """.stripMargin)
 
+    bufVars = null  // explicitly null this field out to allow the referent to be GC'd sooner
```

Comment:
I am curious what happens when `bufVars ` is accessed in `doConsumeWithoutKeys`.

Reply:
The workflow of whole-stage codegen ensures that `doConsumeWithoutKeys` can only be on the call stack when `doProduceWithoutKeys` is also on the call stack; the liveness of the former is strictly a subset of the latter.
That's because for a plan tree that looks like:
```
+- A
  +- B
    +- C
```
The whole-stage codegen system (mostly) works like:
```
A.produce
 |------> B.produce
 |         |------> C.produce
 |         |         |------> B.consume
 |         |         |         |------> A.consume
 |         |         |         |        |
 |         |         |         |<-------o
 |         |         |<--------o
 |         |<--------o
 |<--------o
 o
```

Reply:
Ah, got it. Thank you for the clarification.